test/plugin/test_out_google_cloud.rb

```diff
@@ -1048,6 +1061,18 @@ def test_multiple_cloudfunctions_logs_tag_not_matched
     end
   end
 
+  def test_http_request_from_record
+    setup_gce_metadata_stubs
+    setup_logging_stubs
+    d = create_driver(APPLICATION_DEFAULT_CONFIG)
+    d.emit('httpRequest' => HTTP_REQUEST_MESSAGE)
+    d.run
+    verify_log_entries(1, COMPUTE_PARAMS, 'httpRequest') do |entry|
+      assert_equal HTTP_REQUEST_MESSAGE, entry['httpRequest'], entry
+      assert_equal nil, entry['structPayload']['httpRequest'], entry
+    end
+  end
+
```

Comment:
add a test for the case I mentioned above (httpRequest not a hash) as well as httpRequest with "extra" fields (make sure they make it into the struct payload, or not, depending on what we decide there).